src/source/tcp.rs

```diff
@@ -184,16 +196,17 @@ where
                         let mut handler = H::new();
                         handler.handle_stream(rchans, &rtags, &poller, stream);
                     });
-                    self.handlers.push(new_stream);
+                    handlers.push(new_stream);
                 }
 
                 Err(e) => match e.kind() {
                     std::io::ErrorKind::WouldBlock => {
                         break;
                     }
-                    _ => unimplemented!(),
+                    _ => return Err(e),
```

Comment:
`ConnectionAborted` is an error kind that should be ignored (by continuing) actually. That means a connection arrived in the listen queue but was subsequently closed prior to accept being called.

Reply:
Fixed.

Reply:
Similarly, I believe we should `continue` with `Interrupted` and `TimedOut`

Reply:
Added all of the above.